contracts/core/MarginRequirements.sol

```diff
@@ -0,0 +1,223 @@
+     */
+    modifier onlyOTCWrapper() {
+        require(
+            msg.sender == AddressBookInterface(addressBook).getOTCWrapper(),
+            "MarginRequirements: Sender is not OTCWrapper"
+        );
+
+        _;
+    }
+
+    /**
+     * @notice modifier to check if the sender is the Keeper address
+     */
+    modifier onlyKeeper() {
+        require(
+            msg.sender == AddressBookInterface(addressBook).getKeeper(),
+            "MarginRequirements: Sender is not Keeper"
+        );
+
+        _;
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice sets the initial margin %
+     * @dev can only be called by owner
+     * @param _underlying underlying asset address
+     * @param _collateralAsset collateral asset address
+     * @param _isPut option type the vault is selling
+     * @param _account account address
+     * @param _initialMargin initial margin percentage (eg. 10% = 1000)
+     */
+    function setInitialMargin(
+        address _underlying,
+        address _collateralAsset,
+        bool _isPut,
+        address _account,
+        uint256 _initialMargin
+    ) external onlyOwner {
+        require(
+            _initialMargin > 0 && _initialMargin <= 100 * 10**2,
+            "MarginRequirements: initial margin cannot be 0 or higher than 100%"
+        );
+        require(_underlying != address(0), "MarginRequirements: invalid underlying");
+        require(_collateralAsset != address(0), "MarginRequirements: invalid collateral");
+        require(_account != address(0), "MarginRequirements: invalid account");
+
+        initialMargin[keccak256(abi.encode(_underlying, _collateralAsset, _isPut))][_account] = _initialMargin;
+    }
+
+    /**
+     * @notice sets the maintenance margin absolute amount
+     * @dev can only be called by keeper
+     * @param _account account address
+     * @param _vaultID id of the vault
+     * @param _maintenanceMargin maintenance margin absolute amount
+     */
+    function setMaintenanceMargin(
+        address _account,
+        uint256 _vaultID,
+        uint256 _maintenanceMargin
+    ) external onlyKeeper {
+        require(_maintenanceMargin > 0, "MarginRequirements: initial margin cannot be 0");
+        require(_account != address(0), "MarginRequirements: invalid account");
+
+        maintenanceMargin[_account][_vaultID] = _maintenanceMargin;
+    }
+
+    /************************************************
+     *  MARGIN OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice clears the maintenance margin mapping
+     * @dev can only be called by OTC wrapper contract
+     * @param _account account address
+     * @param _vaultID id of the vault
+     */
+    function clearMaintenanceMargin(address _account, uint256 _vaultID) external onlyOTCWrapper {
```

Comment:
actually, we don't need this method since vaultCounter [incremented](https://github.com/rysk-finance/GammaProtocol/blob/master/contracts/core/Controller.sol#L742) each time vault opened (and not decremented on settle vault

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/be9ff20f8188f6835647c10266bb69672a9cfe5c